docker/pravega/scripts/init_segmentstore.sh

```diff
@@ -13,6 +13,7 @@ init_segmentstore() {
     [ ! -z "$PUBLISHED_ADDRESS" ] && add_system_property "pravegaservice.publishedIPAddress" "${PUBLISHED_ADDRESS}"
     [ ! -z "$PUBLISHED_ADDRESS" ] && add_system_property "pravegaservice.listeningIPAddress" "0.0.0.0"
     [ ! -z "$PUBLISHED_PORT" ] && add_system_property "pravegaservice.publishedPort" "${PUBLISHED_PORT}"
+    [ ! -z "$HOSTNAME" ] && add_system_property "metrics.metricsPrefix" "${HOSTNAME}"
```

Comment:
I like this because it is the right default value for most scenarios.

Note that it is also possible to set this default value directly in Pravega's config file, because typesafe handles envvar substitution.   Frankly that would be better.

Reply:
Controller uses typesafe to load properties, but Segment Store doesn't. I think that we cannot rely on envvar substitution currently, given that the config for metrics is common to both processes. This may be an argument to fix the configuration of Pravega in the future, jointly with other issues that have been raised already (#3059).

Reply:
I don't like very much the idea of setting the metrics prefix implicitly. I'd rather have the application explicitly doing it via an environment variable.

Reply:
I agree with @fpj. I don't see the need to explicitly set the `metricsPrefix` system property based on the `$HOSTNAME` envvar. The application can already set this property via the `JAVA_OPTS` envvar, like:

```
JAVA_OPTS="-Dmetrics.metricsPrefix=pravega-segmentstore-3"
```